src/components/app/routes/data/utils.js

```diff
@@ -346,12 +346,13 @@ export async function ensureActiveEnterpriseCustomerUser({
   }
 
   // Given the user has an active ECU, but the current route has no slug, redirect to the slug of the active ECU.
-  if (activeEnterpriseCustomer && !enterpriseSlug) {
+  if (activeEnterpriseCustomer && activeEnterpriseCustomer.slug !== enterpriseSlug) {
```

Comment:
If we don't need to update the active ECU (per the BFF response), and we're on a BFF-enabled route, we need to only redirect here IFF we're not already on the slug for the `activeEnterpriseCustomer`.

Otherwise, we get into an infinite redirect loop, trying to redirect to its current route it's already on, essentially.